### ISSUE TITLE
kv: TestBeginTransactionErrorIndex needs to be rewritten

### ISSUE THREAD

**Issue by andreimatei** [cockroachdb/cockroach#6408]
Hey Kenji,

This commit
https://github.com/cockroachdb/cockroach/commit/96eca328af8c0e121a8e0e675bf41f53bdd3364a
has introduced this test
https://github.com/cockroachdb/cockroach/blob/c1ba244/client/txn_test.go#L334

I'm about to delete it in https://reviewable.io/reviews/cockroachdb/cockroach/6367#-KGSrupuF7VnPw6V-F7K
because I'm changing the Txn interface to return `error`, not `pErr`, and so there's no longer an index reported with the error. If this test is relevant, I think it needs to either be done at a lower level, or to test another effect at the client level, although this latter option wasn't clear to me how to do exactly. I'm hoping you'd chime in :)

FWIW in the hope it helps, you can see in this PR that the indexes from `pErr` are no longer used by `convertBatchError`, instead we iterate to find the first result with an error.

**Comment by kkaneda** [comment]
Sure, @andreimatei ! Thanks a lot for driving the `error`/`pErr` work!

**Comment by andreimatei** [comment]
Actually nvm. I failed to get rid of that pErr.Index, so I've put the test back. Not sure exactly why I failed, but there code here that smears an error everywhere in the batch results, not just in the result with the corresponding index. I tried to get rid of it, but the panic at line 127 fired, which I don't understand. So I let a `pErr` leave for now in the batch.Results.
https://github.com/cockroachdb/cockroach/blob/149b4fb5ea0f9c7986f0a5f0bd37cb5aac098024/client/batch.go#L118
